docs/webhooks.md

```diff
@@ -0,0 +1,72 @@
+# Webhooks
+
+?> Available `5.2.0`
+
+A variety of webhooks are available to extend the functionality of SQLPad.
+
+To enable a specific webhook, provide a URL for the event you would like to subscribe to. This URL will be sent an HTTP(S) POST with a payload of relevant information when the event occurs.
+
+To ensure a webhook call is valid, `SQLPAD_WEBHOOK_SECRET` may be configured, and validated on every webhook call.
+
+Each webhook will contain the following headers:
+
+- `Content-Type`: `application/json`
+- `SQLPad-Secret`: `SQLPAD_WEBHOOK_SECRET value`
+- `SQLPad-URL`: `http://your-sqlpad-public-url:port/base-url`
+- `SQLPad-Hook-Name`: `hook_name`
+
+The payload sent to the URL varies by event.
+
+## User Created
+
+The user created webhook whenever a user is added via the users API/UI. This hook can be used to automate user invites, sending a message to users inviting them to sign up (if using local user/password authentication) or sign in (if using a single sign on).
+
+**Env**: `SQLPAD_WEBHOOK_USER_CREATED_URL`
+
+**Hook Name**: `user_created`
+
+**Payload Body**:
+
+```js
+{
+  id: '7681aec0-b8c5-4267-b9cb-26f7ff3b3e48',
+  name: 'user1',
+  email: 'user1@test.com',
+  role: 'editor',
+  createdAt: '2020-07-13T19:12:48.020Z'
+}
+```
+
+## Query Created
+
+Fires whenever a query is created (initial query save). This is not fired on query execution (see batch created).
+
+**Env**: `SQLPAD_WEBHOOK_QUERY_CREATED_URL`
+
+**Hook Name**: `query_created`
+
+**Payload Body**:
+
+```js
+{
+  id: '65791206-ae82-4075-acc4-638d0eddf38c',
```

Comment:
again - what is this id? does it make sense to export it?

Reply:
query id - see comment regarding user. Will apply whatever approach is used there for all hooks